ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -250,6 +263,18 @@ object WebSocketService {
           import spray.json._, JsonProtocol._
           "matchedQueries" -> p.toJson
         }
+
+      override def requestOffset(request: SearchForeverRequest): Option[StartingOffset] =
+        request.queries.toList.view
+          .map(_.offset)
+          .fold(none[domain.Offset]) {
+            case (Some(o1), Some(o2)) => Some(domain.Offset.ordering.min(o1, o2))
+            case (o1, None) if o1.isDefined => o1
+            case (None, o2) if o2.isDefined => o2
+            case _ => None
+          }
+          .map(StartingOffset.apply)
```

Comment:
I suspect there is a more principled way to do this, looking forward to suggestions.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9847/commits/ae9d8f7d2dda39f946f5bdac6dc9eb1f0b28b02b